rts/motoko-rts/src/constants.rs

```diff
@@ -3,6 +3,10 @@ use crate::types::{Bytes, Words};
 /// Wasm word size. RTS only works correctly on platforms with this word size.
 pub const WORD_SIZE: u32 = 4;
 
+/// Maximum Motoko array size 2^29 (inclusive)
+/// NB: Must agree with Arr.max_array_size in compile.ml.
+pub const MAX_ARRAY_SIZE: u32 = 1 << 29;
```

Comment:
Technically the 29 is not a magic, but depends on `WORD_SIZE` (above). Not a biggie.

Reply:
I depends on the tagging scheme as well as wordsize. I now have 2 tag bits for plus sign bit, leading to 29.

Reply:
We've lost 1-bit from compact bignums.